BoxContentSDK/BoxContentSDK/Models/BOXRepresentation.m

```diff
@@ -41,27 +46,26 @@ - (instancetype)initWithJSON:(NSDictionary *)JSONResponse
                                                                     nullAllowed:NO];
         self.details = detailsJSON;
         
-        NSDictionary *linksJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyLinks
-                                                                 inDictionary:JSONResponse
-                                                              hasExpectedType:[NSDictionary class]
-                                                                  nullAllowed:NO];
-        
-        NSDictionary *contentJSON = linksJSON[BOXAPIObjectKeyContent];
+        NSDictionary *contentJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyContent
+                                                                   inDictionary:JSONResponse
+                                                                hasExpectedType:[NSDictionary class]
+                                                                    nullAllowed:NO];
         
-        NSString *contentURLString = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyURL
-                                                             inDictionary:contentJSON
-                                                          hasExpectedType:[NSString class]
-                                                              nullAllowed:NO];
+        NSString *contentURLString = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyURLTemplate
+                                                                    inDictionary:contentJSON
+                                                                 hasExpectedType:[NSString class]
+                                                                     nullAllowed:NO];
         if (contentURLString.length > 0) {
+            if ([self.type isEqualToString:BOXRepresentationTypeHLS]) {
+                contentURLString = [contentURLString stringByReplacingOccurrencesOfString:BOXRepresentationTemplateKeyAccessPath withString:BOXRepresentationTemplateValueHLSManifiest];
+            }
             self.contentURL = [NSURL URLWithString:contentURLString];
```

Comment:
I thought I commented on this before. But if you don't do the replacement, this will fail. So, the if check should check for hls and fail if it isn't.